### PR TITLE
Handle VK_NULL_HANDLE attachments in VkRenderingFragmentShadingRateAttachmentInfoKHR

### PR DESCRIPTION
**Pull request by CME42**
Fixes the draw_instanced test in the vkd3d-proton test suite.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by JaxLinAMD** [comment]
ok to test

**Comment by amdvlk-admin** [comment]
### Test summary for commit f9566ba555576bac65a5503e99c1a4e1e9aedd46
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev 5197f7bda7de6b1427a427e70156653eb40525ce</li>
<li>PAL: dev 2483d46fa27c30502e497ea169ee53b142e9fa06</li>
<li>SPVGEN: dev fa39b110650bc72fd7fda54af621e69306036888</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 2a87cb67e0180322832e722db8a4c53a5fc4e867</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/153987)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 38237/67477 (56.7%)</li>
<li>Failed: 0/67477 (0.0%)</li>
<li>Not Supported: 29240/67477 (43.3%)</li><li>Warnings: 0/67477 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 12505/19033 (65.7%)</li>
<li><details><summary>Failed: 1/19033 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.binding_model.buffer_device_address.set3.depth3.baseubo.convertchecku64.store.multi.scalar.frag
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 6527/19033 (34.3%)</li><li>Warnings: 0/19033 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 39235/67477 (58.1%)</li>
<li>Failed: 0/67477 (0.0%)</li>
<li>Not Supported: 28242/67477 (41.9%)</li><li>Warnings: 0/67477 (0.0%)</li>
</ul></details>


</ul></details>

File: icd/api/vk_cmdbuffer.cpp [hunk 1]
```diff
@@ -7509,7 +7509,7 @@ void CmdBuffer::BindTargets(
 
         PalCmdBuffer(deviceIdx)->CmdBindTargets(params);
 
-        if (pRenderingFragmentShadingRateAttachmentInfoKHR != nullptr)
+        if (pRenderingFragmentShadingRateAttachmentInfoKHR != nullptr && pRenderingFragmentShadingRateAttachmentInfoKHR->imageView != VK_NULL_HANDLE)
```
Comment (JonCampbell407) [inline icd/api/vk_cmdbuffer.cpp:7512]: Each expression needs () around it.

The line exceeds the 120 character limit.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3d16a982858725900c0248fd72fbfb15c1b150b2
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev 5197f7bda7de6b1427a427e70156653eb40525ce</li>
<li>PAL: dev 2483d46fa27c30502e497ea169ee53b142e9fa06</li>
<li>SPVGEN: dev fa39b110650bc72fd7fda54af621e69306036888</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 2a87cb67e0180322832e722db8a4c53a5fc4e867</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/202431)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 38237/67477 (56.7%)</li>
<li>Failed: 0/67477 (0.0%)</li>
<li>Not Supported: 29240/67477 (43.3%)</li><li>Warnings: 0/67477 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 38075/67477 (56.4%)</li>
<li>Failed: 0/67477 (0.0%)</li>
<li>Not Supported: 29402/67477 (43.6%)</li><li>Warnings: 0/67477 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 39235/67477 (58.1%)</li>
<li>Failed: 0/67477 (0.0%)</li>
<li>Not Supported: 28242/67477 (41.9%)</li><li>Warnings: 0/67477 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit eef046a7269e33bfb119981791945be843109d1b
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev 5197f7bda7de6b1427a427e70156653eb40525ce</li>
<li>PAL: dev 2483d46fa27c30502e497ea169ee53b142e9fa06</li>
<li>SPVGEN: dev fa39b110650bc72fd7fda54af621e69306036888</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 2a87cb67e0180322832e722db8a4c53a5fc4e867</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/202431)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 38237/67477 (56.7%)</li>
<li>Failed: 0/67477 (0.0%)</li>
<li>Not Supported: 29240/67477 (43.3%)</li><li>Warnings: 0/67477 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 38075/67477 (56.4%)</li>
<li>Failed: 0/67477 (0.0%)</li>
<li>Not Supported: 29402/67477 (43.6%)</li><li>Warnings: 0/67477 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 39235/67477 (58.1%)</li>
<li>Failed: 0/67477 (0.0%)</li>
<li>Not Supported: 28242/67477 (41.9%)</li><li>Warnings: 0/67477 (0.0%)</li>
</ul></details>


</ul></details>

**Review by JonCampbell407 — APPROVED** [review]

**Pull request merged**